helix-term/src/commands/typed.rs

```diff
@@ -1559,6 +1559,42 @@ fn run_shell_command(
     Ok(())
 }
 
+fn read_file_info_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (view, doc) = current!(cx.editor);
+
+    ensure!(!args.is_empty(), "file name is expected");
+
+    let filename = args.get(0).unwrap();
+    let path = PathBuf::from(filename.to_string());
+    if path.exists() {
+        match std::fs::read_to_string(path) {
```

Comment:
I would use `https://docs.rs/encoding_rs/latest/encoding_rs/struct.Decoder.html#method.decode_to_string` using the `doc.encoding` instead.

Reply:
I'd extract parts of the from_reader code and share it.

Reply:
@kirawi Thanks for the pointers! I am having a look.

> I'd extract parts of the from_reader code and share it.

I agree. It seems like `from_reader` is not a direct fit but close enough (i.e. if it worked with a `Tendril` instead of a `Rope`).

Reply:
Using `from_reader` directly is more expensive. I can try to work out something to make the function more generic, but for now I'd recommend copying that function and adapting it for a string.